### DISCUSSION TITLE
More Control questions

### DISCUSSION THREAD

**Discussion by lbeltrame** [vladmandic/sdnext#2756] [Q&A]
As I'm trying things and taking notes more questions come to mind. For the record, I 'll share my findings with @Aptronymist in one way or the other once I'm done.
Today's question is about multiple control images. Is this workflow supported? An example I use when doing complex poses involves multiple ControlNets and often manually adjusted preprocessed images.

Example:

I want to generate a gymnast with a pole. I usually create the pose with a 3D mannequin externally (e.g. in CLIP STUDIO PAINT) , then generate the Openpose bones (with the Original backend, this involves opening the Openpose editor and manually adjusting things like face outlines and fingers). Then I make lineart with the appropriate preprocessor, and manually erase all but the hands and the pole. Lastly I make a depth image with either midas or leres++.
To save computation time I then upload the preprocessed images along with the modified lineart, select the right model for them, and use the "none" preprocessor for each of them.
After that is done, I input my prompt and generate the image(s).

As you can see this workflow requires three separate control images, but in certain cases I've gone up to 4 (segmentation, often manually adjusted). So the 5 cent question is: is this possible with Control? I see that you can upload images for each ControlNet you add but it doesn't seem to affect the image generation.

**Comment by vladmandic** [comment]
that's exactly the purpose of the option to upload image to each unit - if that doesn't work, i'll have to check why.

**Reply by lbeltrame** [reply]
I have plenty of time to test things this weekend - I'll test things and report back. At least with OpenPose it seemed to lock me into 512x512 (with no initial control image) because the first resize would be ignored, *and* the pose would get ignored. But I did a one-off test so I need to repeat it.
If I find something that doesn't work I'll try to file an issue with a MWE (and perhaps the used control image) so that it can be independently tested.
